clang/docs/SYCLFERFCIntegrationHeaderFooter.md

```diff
@@ -0,0 +1,179 @@
+# Integration Header
+
+We propose an Integration Header for two purposes:
+
+* to allow a non-SYCL-aware compiler to do host compilations and
+* as a mechanism permitting the SYCL run-time library to use the information provided for its needs
+
+Consider the following partial code:
+
+```
+  sycl::accessor<char, 1, sycl::access::mode::read> acc1;
+  int i = 13;
+  sycl::sampler smplr;
+  struct S {
+    char c;
+    int i;
+  } test_s;
+  test_s.c = 14;
+  kernel_single_task<class first_kernel>([=]() {
+    if (i == 13 && test_s.c == 14) {
```

Comment:
Pedantic-ness if speak about the type trait (otherwise it can be understood as a local type).

```suggestion
  class first_kernel;
[...]
  sycl::accessor<char, 1, sycl::access::mode::read> acc1;
  int i = 13;
  sycl::sampler smplr;
  struct S {
    char c;
    int i;
  } test_s;
  test_s.c = 14;
  kernel_single_task<class first_kernel>([=]() {
    if (i == 13 && test_s.c == 14) {
```